src/controllers/profile.js

```diff
@@ -1,21 +1,20 @@
-'use strict';
-
-const express = require('express');
-const fetch = require('node-fetch');
-const ArsenalParser = require('@wfcd/arsenal-parser');
-const flatCache = require('flat-cache');
-const path = require('path');
-
-const { noResult, cache } = require('../lib/utilities');
+import express from 'express';
+import ArsenalParser from '@wfcd/arsenal-parser';
+import flatCache from 'flat-cache';
+import { dirname, resolve } from 'node:path';
+import { fileURLToPath } from 'node:url';
+import { noResult, cache } from '../lib/utilities.js';
 
 const router = express.Router({ strict: true });
 
 const WF_ARSENAL_ID = 'ud1zj704c0eb1s553jbkayvqxjft97';
 const WF_ARSENAL_API = 'https://content.warframe.com/dynamic/twitch/getActiveLoadout.php';
 let token;
 
+const DIRNAME = dirname(fileURLToPath(import.meta.url));
```

Comment:
I feel kinda bad for picking on you but DIRNAME isn't constant as it's value is assigned at runtime, everything in js is so there's no true constant, but something like the arsenal id could be, and also you never really know where the program is being executed at compile time it's a runtime thing by design.

Some alternatives that i can think about `dirPath` `currDir` `currentDirectory` `folderPath`

Reply:
yeah, sorry, that's dumb

Reply:
if you didn't notice the other 2 consts in the file....

Reply:
sorry didn't quite understand if DIRNAME is dumb or renaming it to something else is dumb

Reply:
having an issue with an actual const that's determined once at compile time when the app starts (given that's how files are cached in esm), not constantly at runtime

Reply:
are you referring to `WF_ARSENAL_API` and `WF_ARSENAL_ID`? Because i'm still confused

Reply:
or maybe you're discussing about what a constant value means in js? even with chatgpt i did not understand it

Reply:
DIRNAME, in esm, is defined once in the file, and never again. unless the request for the file has a different query string or is cache-busted, the value won't be re-evaluated, and is thus constant.

Reply:
If non re-evaluation of the top level declared variables in js is the definition of constant values, a bunch of things would also be considered like that.

For example:
```js
export const platforms = ['pc', 'ps4', 'xb1', 'swi'];
export const solKeys = Object.keys(wfData.solNodes);
```

Since most people derive this concept of constant values to js from languages like C, my definition is if js was a compiled language what values could certainly be known at the compilation time? Basically it needs to be deeply immutable and immutability it's not guaranteed when parsing data with the function dirname and fileUrlToPath, which will probably look odd for anyone used to rust or C.

While this discussion is mostly interesting, i'm able to compromise and have a special case for DIRNAME or use your definition of constant in this project.

https://google.github.io/styleguide/jsguide.html#naming-constant-names
https://developer.mozilla.org/en-US/docs/Glossary/Constant
https://en.wikipedia.org/wiki/Constant_(computer_programming)

Reply:
quite simply, this roudabout of what to call the dirname is pointless. the longstanding standard is `__dirname`. i'm trying to preserve the standard format much as i've been able to.